src/shogun/base/SGObject.h

```diff
@@ -857,6 +862,13 @@ class CSGObject
 	{
 		BaseTag tag(name);
 		create_parameter(tag, AnyParameter(make_any_ref(value), properties));
+		
+		if constexpr (is_sg_base<T>::value) {
```

Comment:
imo if you wrap this whole block into a constexpr function then you could just call that instead of copy pasting things 4 times.